app/main.js

```diff
@@ -469,37 +474,70 @@ function onionSkinAmount() {
     document.getElementById("onionSkinPercentage").innerHTML = document.getElementById("onionSkinAmount").value * 5 + "%";
     onionSkinWindow.style.opacity = (document.getElementById("onionSkinAmount").value * 5)/100;
 }
+
 /**
  * Set directory to export frames to
  */
 function checkdefaultdirectory() {
-    frameExportDirectory = localStorage.getItem('default_directory');
+    frameExportDirectory = _getDefaultDirectory();
     if (frameExportDirectory === null) {
         console.log("no default set");
     } else {
-        document.getElementById("currentDirectoryName").innerHTML = frameExportDirectory;
-        console.log("default directory found: " + frameExportDirectory);
-        document.title = "Boats Animator (" + frameExportDirectory + ")";
+        _displayDirectory(frameExportDirectory);
     }
 }
- function chooseFile(name) {
+
+/**
+ * Open the system native choose directory dialog.
+ *
+ * @param {String} The DOM selector to the dialog trigger.
+ */
+function chooseFile(name) {
     var chooser = document.querySelector(name);
-     chooser.addEventListener("change", function(evt) {
-         console.log("Directory set as " + this.value);
-         frameExportDirectory = this.value;
-         document.getElementById("currentDirectoryName").innerHTML = frameExportDirectory;
-         document.title = "Boats Animator (" + frameExportDirectory + ")";
-    }, false);
 
-    chooser.click();  
-  }
-  
-function changedirectory() {
-    chooseFile('#chooseDirectory'); 
+    chooser.addEventListener("change", function() {
+        frameExportDirectory = this.value;
+        _displayDirectory(frameExportDirectory);
+    });
+
+  chooser.click();
 }
-function setdefaultdirectory() {
-    localStorage.setItem("default_directory",frameExportDirectory);
+
+/**
+ * Display the frame destination directory in the UI.
+ *
+ * @param {String} dir The directory to display.
+ */
+function _displayDirectory(dir) {
+    console.log(`Current destination directory is ${dir}`);
+    document.getElementById("currentDirectoryName").innerHTML = dir;
+    document.title = `Boats Animator (${dir})`;
```

Comment:
This little bit right here is an [ES6 template string](http://davidwalsh.name/javascript-template-strings). They are really cool but sadly can't used full-time on the web. However, because BA is running in an NW.js wrapper that uses the latest Blink and V8 engines that power Chrome, we are use them all we want. Cool, huh? :smiley: